lumicks/pylake/kymo.py

```diff
@@ -281,7 +281,12 @@ def from_dataset(cls, h5py_dset, file):
         start = h5py_dset.attrs["Start time (ns)"]
         stop = h5py_dset.attrs["Stop time (ns)"]
         name = h5py_dset.name.split("/")[-1]
-        json_data = json.loads(h5py_dset[()])["value0"]
+        try:
+            json_data = json.loads(h5py_dset[()])["value0"]
+        except KeyError:
+            warnings.warn(f"Scan '{name}' is missing metadata and cannot be loaded")
+            return None
```

Comment:
An argument can be made here that his should actually re-raise the exception with more information, rather than translate that exception into a `None` result:
```py
        try:
            json_data = json.loads(h5py_dset[()])["value0"]
        except KeyError:
            raise KeyError(f"Scan '{name}' is missing metadata and cannot be loaded")
```

```py
        def try_from_dataset(*args):
            try:
                return Scan.from_dataset(*args)
            except Exception as e:
                warnings.warn(str(e))
                return None

        scan_objects = [(name, try_from_dataset(dset, self)) for name, dset in self.h5["Scan"].items()]
        return {name: scan for name, scan in scan_objects if scan is not None}
```

There are two reasons:
1. Returning `None` for a specific set of errors means that we get one set of errors the usual way (exceptions) and the other as `None` results. However, this is pretty surprising. We usually expect to get all errors the same way unless there's a really good reason -- in which case it should be mentioned in the docstring (it's part of the API and error handling changes are also API breaking changes).
2. Previously, `Scan.from_dataset()` would always return a proper object or raise an exception. Now it can return `None` which means that it has `Optional` semantics. Since `.from_dataset()` is a constructor this can imply that type itself has optional semantics. If that's true, it shifts the burden to users to check `if obj is None`.

Essentially:
```py
try:
    scan = Scan.from_dataset(...)
    if scan is not None:
        # good path
    else:
        # failure path 1
except:
    # failure path 2
```
Having two failure paths is sometime justified, but it's more of a thing where you need to explicitly prove "yes, this is the best API for this type". It should not be done accidentally because it does increase the API area and it adds more stuff to document and more stuff for users to learn.

Reply:
Good point!

Reply:
I see your point here. One of the issues is that if `Scan.from_dataset()` throws an exception, then any code requesting the list of available objects will break (`File.__str__()` and `File.scans`) Maybe the best thing to do is throw the exception from `from_dataset()` and handle that in the other two methods? Should be a pretty simple fix. What do you think?

Reply:
which now re-reading your suggestion is exactly what I just said...sorry. I'll go ahead and change it up

Reply:
Indeed, that's what I was suggesting with the `try_from_dataset()` helper function, but I left it a bit obscure as the example code.